react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,15 +116,55 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => boolean;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {boolean}
+     */
+    _onFinalizeServerURL() {
+        let { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(\\w+(?:s)?://)?(.+)$');
+            const urlComponents = urlRegExp.exec(serverURL.toLowerCase());
+
+            if (!urlComponents[1] || !urlComponents[1].startsWith('http')) {
+                serverURL = `https://${urlComponents[2]}`;
+            }
+
+            const parsedURI
+                = parseStandardURIString(serverURL);
+
+            if (!parsedURI.host) {
+                return false;
+            }
+
+            this._onChangeServerURL(parsedURI.toString());
+        }
+
+        return true;
+    }
+
     _onRequestClose: () => void;
 
     /**
      * Handles the back button.
+     * Also invokes _onFinalizeServerURL because field events won't
+     * trigger if the user exits the screen while editing.
      *
      * @returns {void}
      */
     _onRequestClose() {
-        this.props.dispatch(hideAppSettings());
+        if (this._onFinalizeServerURL()) {
+            this.props.dispatch(hideAppSettings());
+        } else {
+            this.showURLAlert();
```

Comment:
This logic should be in the native part, there is currently no web part, but when we have one , validation would probably go inline, since there is space for it.